projects/storefrontlib/src/cms-components/cart/validation/cart-validation-warnings-state.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { skip, take, tap } from 'rxjs/operators';
+import { RoutingService, CartModification } from '@spartacus/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CartValidationWarningsStateService {
+  constructor(protected routingService: RoutingService) {
+    this.routingService
+      .getRouterState()
+      .pipe(take(1))
+      .subscribe(
+        (routerState) => (this.navigationIdCount = routerState.navigationId)
+      );
+  }
+  NAVIGATION_SKIPS = 2;
+  navigationIdCount: number;
+
+  cartValidationResult$ = new Subject<CartModification[]>();
+  checkoutRouteActivated$ = new BehaviorSubject<boolean>(false);
+
+  checkForValidationResultClear$ = this.routingService.getRouterState().pipe(
+    skip(this.NAVIGATION_SKIPS),
+    tap((routerState) => {
+      if (
+        this.navigationIdCount + this.NAVIGATION_SKIPS <=
+        routerState.navigationId
+      ) {
+        this.cartValidationResult$.next([]);
+        this.navigationIdCount = routerState.navigationId;
+      }
```

Comment:
In this case, once you clean the state, again after 2 navigations you will clear it once again and so on. I would wrap it with a check that verifies if the state is not empty (array length probably) and only then do the clear.